src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/TextsController.cs

```diff
@@ -36,27 +38,34 @@ public TextsController(IText text)
         [HttpGet]
         public async Task<ActionResult<TextResource>> Get(string org, string app, [FromRoute] string language)
         {
-            TextResource textResource;
-
             if (!string.IsNullOrEmpty(language) && language.Length != 2)
             {
                 return BadRequest($"Provided language {language} is invalid. Language code should consists of two characters.");
             }
 
-            textResource = await _text.GetText(org, app, language);
-            if (textResource != null)
+            byte[] textsAsByteArray = _appResources.GetText(org, app, $"resource.{language}.json");
```

Comment:
Consider moving all this logic from line 46 to line 66 into the AppResourcesSI class and have it return the TextResource with a simple get.

Reply:
Inspired by your comment I made some changes. The existing GetText method is used, but I made a new method.